src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/TearSheetLinksTransformer.java

```diff
@@ -1,86 +1,72 @@
-                    // names to be all lower case, makes the handler less
-                    // dependent on processor order
-                    String id = StringUtils.isNotBlank(el.getAttribute("CompositeId")) ? el.getAttribute("CompositeId")
-                            : el.getAttribute("compositeid");
-                    if (StringUtils.isNotBlank(id)) {
-                        builder.queryParam("identifierValue", URLEncoder.encode(id, "UTF-8"));
-                    }
+            for (int i = len - 1; i >= 0; i--) {
+                Element el = (Element) nodes.item(i);
+                // this is because the previous processor changes attribute
+                // names to be all lower case, makes the handler less
+                // dependent on processor order
+                String id = StringUtils.isNotBlank(el.getAttribute("CompositeId")) ? el.getAttribute("CompositeId")
+                        : el.getAttribute("compositeid");
+                if (StringUtils.isNotBlank(id)) {
+                    identifierValues.add(id);
                 }
-
-            } catch (UnsupportedEncodingException e) {
-                throw new RuntimeException("Encoding for concordance call failed  ");
             }
-
-            URI concordanceApiQuery = builder.buildFromEncoded();
-            Concordances responseConcordances = client.resource(concordanceApiQuery)
-                    .header("Host", "public-concordances-api").get(Concordances.class);
-            if (responseConcordances != null && responseConcordances.getConcordances() != null
-                    && !responseConcordances.getConcordances().isEmpty()) {
-
-                transformTearSheetLink(responseConcordances.getConcordances(), nodes);
-            } else {
-                List<String> identifiers = URLEncodedUtils.parse(concordanceApiQuery, "UTF-8").stream()
-                        .filter(item -> item.getName().equals("identifierValue")).map(NameValuePair::getValue)
-                        .collect(Collectors.toList());
-                identifiers.forEach(item -> LOG.warn("Composite Id is not concorded CompositeId=" + item));
+            try {
+                Concordances concordances = client.getConcordancesByIdentifierValues(identifierValues);
+                if (concordancesArePresent(concordances)) {
+                    transformTearSheetLink(concordances.getConcordances(), nodes);
+                } else {
+                    identifierValues.forEach(item -> LOG.warn("Composite Id is not concorded CompositeId=" + item));
+                }
+            } catch (ConcordanceApiException e) {
+                LOG.warn("Unable to retrieve concordances for identifier values: " + identifierValues, e);
             }
         }
     }
 
+    private boolean concordancesArePresent(Concordances concordances) {
+        return concordances != null && concordances.getConcordances() != null && !concordances.getConcordances().isEmpty();
```

Comment:
Also here, you could use Strings.isNullOrEmpty for concordances.getConcordances()

Reply:
same.